src/cmd.fs

```diff
@@ -12,11 +12,11 @@ open System
 /// Dispatch - feed new message into the processing loop
 type Dispatch<'msg> = 'msg -> unit
 
-/// Subscription - return immediately, but may schedule dispatch of a message at any time
-type Sub<'msg> = Dispatch<'msg> -> unit
+/// Effect - return immediately, but may schedule dispatch of a message at any time
+type Effect<'msg> = Dispatch<'msg> -> unit
```

Comment:
The name is open of debate, from elmish perspective what's important is that it is a source of messages to be fed into the dispatch loop. Maybe `Feed`?
We also talked about cognitive overload of dealing with aliases. Yes, they make it easy to deal with functions, but they also make it harder to diagnose the errors compiler might give you.

Reply:
Looking at this abstraction from the source material Elm, each item in a Cmd list ends up as a `Task error msg`, which represents a single side effect and its return msg. The F# version of that would probably be `Async<'msg>`[^1].

An Elm Sub item turns into a Process, sortof an MVU app minus the view. It stays "running" until killed. Perhaps represented more economically in F# as `{ Start: unit -> IDisposable * IObservable<'msg> }`.

Elmish's msg return mechanism is dispatch. Which expands the scope of what this abstraction (v3 Sub) does vs in Elm. The user can share out it's dispatch fn, by necessity making dispatch multi-use. So it's not quite a Cmd item because it doesn't represent a single effect and its return value.[^2] And it's not quite a Sub item because you have to explicitly manage its lifecycle side effects if you need other than app lifetime. It's made due for both concerns up to now. With v4 Subs in place, it wouldn't have to.

[^1]: To represent failure, the Msg case can be `of Result<>`, or different cases can be returned for failure/success. Elm uses Tasks for all side effects. But they have a special `Never` type to represent when a DU cases is impossible. So `Task Never msg` is roughly equivalent to `Async<'msg>`.

[^2]: The list wrapper provided by Elmish.Cmd is not technically needed. Multiple v3 Subs can be combined by wrapping in a fan-out dispatch fn. `let batch subs = fun dispatch -> subs |> Seq.iter (fun sub -> sub dispatch)`